--- a/index.php
+++ b/index.php
@@ -43,6 +43,10 @@
         $totalWeight = $unitWeight * $quantity; // El peso total siempre estará en gramos
         $ingredients = [];
 
+        $totalFlour = 0;
+        $totalHydration = 0;
+        foreach ($recipe['ingredients'] as $ingredient => $percentage) {
+          if(isset($recipes['ingredients'][$ingredient]['percentage'])) $totalFlour += ($totalWeight * ($percentage / 100)) * ($recipes['ingredients'][$ingredient]['percentage'] / 100);
+        }
         // Calcular la cantidad de cada ingrediente en el prefermento
         $prefermentIngredients = [];
         if ($prefermentPercentage > 0) {
@@ -50,10 +54,10 @@
 
             // Detectar harina alternativa y leche
             $alternativeFlour = null;
-            $liquid = 'Agua'; // Por defecto se usa agua
+            $liquid = null; // Por defecto se usa agua
             foreach ($recipe['ingredients'] as $ingredient => $percentage) {
-                if (stripos($ingredient, 'harina') !== false && $ingredient !== 'Harina') {
-                    $alternativeFlour = $ingredient;
+                if (isset($recipes['ingredients'][$ingredient]['percentage']) && $recipes['ingredients'][$ingredient]['percentage'] > 0 && $ingredient !== 'Harina de trigo') {
+                  $alternativeFlour = $ingredient;
                 }
                 if ($ingredient === 'Leche') {
                     $liquid = 'Leche';
@@ -61,12 +65,16 @@
             }
 
             // Calcular los ingredientes del prefermento
-            $prefermentFlour = $prefermentWeight * 0.5; // 50% de harina en el prefermento
+            $prefermentFlour = ($prefermentWeight / 2);
             $prefermentLiquid = $prefermentWeight * 0.5; // 50% de líquido en el prefermento
+
             if ($alternativeFlour) {
-                $prefermentIngredients[$alternativeFlour] = round($prefermentFlour, 2);
+                $prefermentIngredients[$alternativeFlour] = $prefermentFlour * ($recipes['ingredients'][$alternativeFlour]['percentage'] / 100);
             } else {
-                $prefermentIngredients['Harina'] = round($prefermentFlour, 2);
+                $prefermentIngredients['Harina de trigo'] = $prefermentFlour;
             }
+
+            if(!$liquid) $liquid = "Agua";
             $prefermentIngredients[$liquid] = round($prefermentLiquid, 2);
             $prefermentIngredients['Levadura'] = round($totalWeight * ($recipe['ingredients']['Levadura'] / 100), 2); // 100% de levadura en el prefermento
         }
@@ -79,11 +87,19 @@
             }
 
             $ingredientWeight = round($totalWeight * ($percentage / 100), 2);
+            if(isset($recipes['ingredients'][$ingredient]['percentage']) && $recipes['ingredients'][$ingredient]['percentage'] > 0){
+                $ingredientWeight = $totalFlour * ($recipes['ingredients'][$ingredient]['percentage'] / 100);
+            }
             if (isset($prefermentIngredients[$ingredient])) {
                 $ingredientWeight -= $prefermentIngredients[$ingredient];
             }
+            if(isset($recipes['ingredients'][$ingredient]['hydrates']) && $recipes['ingredients'][$ingredient]['hydrates'])
+              $totalHydration += $ingredientWeight;
+
             $ingredients[$ingredient] = $ingredientWeight;
         }
+
+        $hydrationPercentage = ($totalHydration / $totalFlour) * 100;
 
         $_SESSION['calculated_recipe'] = [
             'name' => $selectedRecipe,
@@ -91,7 +107,8 @@
             'quantity' => $quantity,
             'total_weight' => round($totalWeight, 2),
             'ingredients' => $ingredients,
-            'preferment' => $prefermentIngredients // Agregar el prefermento a la sesión
+            'preferment' => $prefermentIngredients, // Agregar el prefermento a la sesión
+            'hydration' => round($hydrationPercentage,2)
         ];
 
         header('Location: index.php');
@@ -168,6 +185,10 @@
                         <span class="font-medium"><strong>Peso Total:</strong></span>
                         <span><?= $calculatedRecipe['total_weight'] ?> g</span>
                     </div>
+                    <div class="mb-2">
+                        <span class="font-medium"><strong>Hidratación:</strong></span>
+                        <span><?= $calculatedRecipe['hydration'] ?> %</span>
+                    </div>
                     <?php
                     // Calcular el coste de la receta (opcional)
                     $totalCost = 0; // Inicializar el coste total